File: backend/src/main/java/com/ntg/sms/Entities/Dtos/Response/AssignmentResponse.java
package com.ntg.sms.Entities.Dtos.Response;

import lombok.Builder;
import lombok.Data;

import java.time.LocalDate;
import java.util.Set;

@Data
@Builder
public class AssignmentResponse {
    private int completed;
    private int total;
    private int completionPercent;
    private Long id;
    private String name;
    private LocalDate deadline;
    private LocalDate assignDate;
    private String description;
    private String fileLink;
    private String studentSubmission;
    private Set<Long> courseIds;

}

File: Frontend/Student/src/app/models/assignment-response.ts
export interface AssignmentResponse {
  id: number;
  name: string;
  deadline: string;        // LocalDate  'YYYY-MM-DD'
  assignDate: string;
  description: string;
  fileLink: string;
  studentSubmission: string;
  courseIds: number[];
  completed: number;
  total: number;
  completionPercent: number;
}